### PR TITLE
refactor: use createRequire to load/resolve modules

### PR DESCRIPTION
**Pull request by sodatea**
See https://github.com/benmosher/eslint-plugin-import/pull/1591 and https://github.com/benmosher/eslint-plugin-import/pull/1602

As far as I understand, `createRequire` is the only semantically correct way to resolve modules from a different context.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Docs
- [ ] Underlying tools
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Other information:**

### CODE REVIEW

**Comment by sodatea** [comment]
@arcanis Can this change eliminate the need to move Generator logic into project local dependencies as said in https://github.com/vuejs/vue-cli/issues/4531?

File: packages/@vue/cli-shared-utils/lib/module.js [hunk 1]
```diff
@@ -35,9 +51,13 @@ const resolve = semver.satisfies(process.version, '>=10.0.0')
 exports.resolveModule = function (request, context) {
   let resolvedPath
   try {
-    resolvedPath = resolve(request, {
-      paths: [context]
-    })
+    try {
+      resolvedPath = createRequire(context).resolve(request)
```
Comment (arcanis) [inline packages/@vue/cli-shared-utils/lib/module.js]: Is `context` a file, or a directory? I think `createRequire` only takes files, so worst case you can `join` it to a fake filename.

**Comment by arcanis** [comment]
@sodatea I improved Yarn in https://github.com/yarnpkg/berry/pull/630 to support the use case of Vue-CLI (and CRA), so the problem we described in #4531 should be fixed!

That said the changes in this PR sound for the best regardless because of the post I shared you on Twitter (https://github.com/yarnpkg/berry/issues/693#issuecomment-575389842) 👍

File: packages/@vue/cli-shared-utils/lib/module.js (hunk 1 above)
Comment (sodatea) [reply packages/@vue/cli-shared-utils/lib/module.js]: 👍It's a directory. I've got it fixed now

**Comment by arcanis** [comment]
By any chance, can `master` be released? I'll be releasing the very first stable build for Yarn 2 tomorrow morning 😊

**Comment by sodatea** [comment]
That soon?
OK, I'll try to squeeze some time testing it and release a new version in hours. It's Chinese New Year's Eve today so I didn't expect a new release… But a patch is fine.

**Comment by arcanis** [comment]
Oh I'm sorry, I should have told you about the release earlier - please disregard, it's not that urgent (people will be able to add your Vue plugin anyway).

Happy new year!

**Pull request merged**